internal/db/schema/state.go

```diff
@@ -0,0 +1,57 @@
+package schema
+
+import (
+	"context"
+)
+
+const nilVersion = -1
+
+type migrationState struct {
+	// devMigration is true if the database schema that would be applied by
+	// InitStore would be from files in the /dev directory which indicates it would
+	// not be safe to run in a non dev environment.
+	devMigration bool
+
+	// binarySchemaVersion provides the database schema version supported by
+	// this binary.
+	binarySchemaVersion int
+
+	upMigrations   map[int][]byte
+	downMigrations map[int][]byte
+}
+
+var migrationStates = make(map[string]migrationState)
+
+// State contains information regarding the current state of a boundary database's schema.
+type State struct {
+	InitializationStarted bool
+	Dirty                 bool
+	CurrentSchemaVersion  int
+	BinarySchemaVersion   int
+}
+
+// State provides the state of the boundary schema contained in the backing database.
+func (b *Manager) State(ctx context.Context) (*State, error) {
```

Comment:
why have this receiver function here and not in manager.go where I'd sort of expect it?

Reply:
Moved CurrentState to manager.go along with the test.